state/txindex/kv/kv.go

```diff
@@ -123,11 +99,6 @@ func (txi *TxIndex) Index(result *abci.TxResult) error {
 	// index tx by events
 	txi.indexEvents(result, hash, b)
 
-	// index tx by height
-	if txi.indexAllEvents || tmstring.StringInSlice(types.TxHeightKey, txi.compositeKeysToIndex) {
-		b.Set(keyForHeight(result), hash)
-	}
-
 	// index tx by hash
 	rawBytes, err := proto.Marshal(result)
```

Comment:
@melekes it seems we automatically were indexing by hash it wasn't turned off if I'm not reading this incorrectly?